lib/DeonData.ts

```diff
@@ -192,9 +192,9 @@ export const mkIntValue = (n: number): IntValue =>
 
 export interface FloatValue {
   class: 'FloatValue';
-  d: number;
+  d: string;
 }
-export const mkFloatValue = (n: number): FloatValue =>
+export const mkFloatValue = (n: string): FloatValue =>
```

Comment:
Maybe we should do some validation of the string here? Throw an exception upon creation if it is not a valid number

Reply:
Something ala https://stackoverflow.com/questions/175739/built-in-way-in-javascript-to-check-if-a-string-is-a-valid-number perhaps

Reply:
So `let isNumeric = s => s !== '' && !isNaN(s)`? In that case I think we should also check that the parameter in `mkIntValue` is `int32`.

Reply:
Fine by me :+1:

Reply:
Actually we should also check if it's finite. And that is kinda difficult without parsing the number ourselves. `isFinite(+"1e309") === false` but the range for decimal128 is all the way up to `1e6144`. So I don't think we should do this.